airflow/jobs.py

```diff
@@ -1289,15 +1299,32 @@ def _execute_task_instances(self,
         """
         executable_tis = self._find_executable_task_instances(simple_dag_bag, states,
                                                               session=session)
-        tis_with_state_changed = self._change_state_for_executable_task_instances(
-            executable_tis,
-            states,
-            session=session)
-        self._enqueue_task_instances_with_queued_state(
-            simple_dag_bag,
-            tis_with_state_changed)
-        session.commit()
-        return len(tis_with_state_changed)
+        if self.max_tis_per_query == 0:
```

Comment:
Can we deduplicate this logic? i.e. tis_per_query == 0 means chunks just has one element (then reuse the rest of the logic.